storehaus-redis/src/main/scala/com/twitter/storehaus/redis/RedisSetStore.scala

```diff
@@ -38,34 +38,34 @@ object RedisSetStore {
  * A Store for sets of values backed by a Redis set.
  */
 class RedisSetStore(val client: Client, ttl: Option[Duration])
-  extends Store[ChannelBuffer, Set[ChannelBuffer]] {
+  extends Store[Buf, Set[Buf]] {
 
-  override def get(k: ChannelBuffer): Future[Option[Set[ChannelBuffer]]] =
+  override def get(k: Buf): Future[Option[Set[Buf]]] =
     client.sMembers(k).map {
       case e if e.isEmpty => None
       case s => Some(s)
     }
 
-  override def put(kv: (ChannelBuffer, Option[Set[ChannelBuffer]])): Future[Unit] =
+  override def put(kv: (Buf, Option[Set[Buf]])): Future[Unit] =
     kv match {
       case (k, Some(v)) =>
-        client.del(Seq(k)) // put, not merge, semantics
-        ttl.map(exp => client.expire(k, exp.inSeconds))
+        client.dels(Seq(k)) // put, not merge, semantics
```

Comment:
we don't need to wrap with a `Seq`. Could we just call `client.del(k)` maybe? Also, this seems like a a race to me. the del could happen after the set. We should use flatMap to sequence:
```scala
for {
  _ <- client.del(k)
  _ <- ttl.fold(Future.Unit) { exp => client.expire(k, exp.inSeconds) }
  _ <- set(k, v.toList)
} yield ()
```